### DISCUSSION TITLE
Auto merge and rebase approved PRs

### DISCUSSION THREAD

**Discussion by romain-sen** [renovatebot/renovate#33208] [Request Help] [answered]
### How are you running Renovate?

A Mend.io-hosted app

### Please tell us more about your question or problem

I have a classic configuration of Renovate, I think, and it does the job. But Renovate does not merge PRs automatically; I need to approve them manually.
So when a new dependency is available, Renovate creates a PR, the CI runs, and then, until I approve it, the PR is rebased again and again each time something is merged into main. Also, PRs for major dependencies that require some work, need to wait for a patch, or cannot be merged immediately stay open and are rebased repeatedly, costing my team some CI time.

We were thinking that we could create Renovate PRs as drafts so the CI doesn't run, and once approved, it does. But we can't approve a PR if we're not sure the CI passes in the first place.

The ideal solution, I think, would be for Renovate to create a PR with the new dependency. The CI runs, and we can see if it's okay to merge or not. Renovate does not rebase it until someone approves the PR, and once approved, it rebases it and merges it. But is it even possible to tell Renovate to rebase only approved PRs?

Here is my config :

```
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:recommended"
  ],
  "rebaseWhen": "auto",
  "packageRules": [
    {
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "automerge": true
    },
    {
      "matchDepTypes": [
        "devDependencies"
      ],
      "automerge": true
    }
  ],
  "platformAutomerge": true
}
```

Thank you for your help !

**Comment by RahulGautamSingh** [comment, marked as answer]
https://docs.renovatebot.com/configuration-options/#keepupdatedlabel

**Reply by romain-sen** [reply]
can I still have renovate to automerge the PRs that are labelled with keepUpdated without the rebase of others ?

**Reply by romain-sen** [reply]
works exactly as I wanted thank you !

**Comment by rarkins** [comment]
If you configure Renovate to automerge, it will automatically keep the branch updated with the base branch. If you turn off automerge, you should find it only rebases when conflicted.
